pkg/ipamclaimswebhook/podmutator.go

```diff
@@ -233,54 +276,94 @@ func ensureIPAMClaimRefAtNetworkSelectionElements(ctx context.Context,
 	return hasChangedNetworkSelectionElements, nil
 }
 
-func findNewPrimaryNetworkIPAMClaimName(ctx context.Context,
-	cli client.Client, pod *corev1.Pod, vmName string) (string, error) {
+func findPrimaryUDNLogicalNetworkAttachment(ctx context.Context,
+	cli client.Client, vmi *virtv1.VirtualMachineInstance) (*virtv1.Interface, error) {
 	log := logf.FromContext(ctx)
-	if pod.Annotations[config.OVNPrimaryNetworkIPAMClaimAnnotation] != "" {
-		return "", nil
-	}
-	primaryNetworkNAD, err := udn.FindPrimaryNetwork(ctx, cli, pod.Namespace)
+	primaryNetworkNAD, err := udn.FindPrimaryNetwork(ctx, cli, vmi.Namespace)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	if primaryNetworkNAD == nil {
-		return "", nil
+		return nil, nil
 	}
 	pluginConfig, err := config.NewConfig(primaryNetworkNAD.Spec.Config)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	if !pluginConfig.AllowPersistentIPs {
-		return "", nil
+		return nil, nil
 	}
 
 	log.Info(
 		"will request primary network persistent IPs",
 		"NAD", client.ObjectKeyFromObject(primaryNetworkNAD),
 		"network", pluginConfig.Name,
 	)
+
+	podNetwork := vmiPodNetwork(vmi)
+	if podNetwork == nil {
+		log.Info(
+			"vmi has no pod network primary UDN ipam claim will not be created",
+			"vmi",
+			client.ObjectKeyFromObject(vmi),
+		)
+		return nil, nil
+	}
+
+	return vmiPodPrimaryNetworkInterface(vmi, *podNetwork), nil
+}
+
+func runningVMForPod(pod *corev1.Pod, vmName string, cli client.Client) (*virtv1.VirtualMachineInstance, error) {
 	vmKey := types.NamespacedName{Namespace: pod.Namespace, Name: vmName}
 	vmi := &virtv1.VirtualMachineInstance{}
 	if err := cli.Get(context.Background(), vmKey, vmi); err != nil {
-		return "", err
+		return nil, err
 	}
-
-	networkName := vmiPodNetworkName(vmi)
-	if networkName == "" {
-		log.Info("vmi has no pod network primary UDN ipam claim will not be created", "vmi", vmKey.String())
-		return "", nil
-	}
-
-	return claims.ComposeKey(vmi.Name, networkName), nil
+	return vmi, nil
 }
 
-// returns the name of the kubevirt VM pod network
-func vmiPodNetworkName(vmi *virtv1.VirtualMachineInstance) string {
+// returns the KubeVirt VM pod network
+func vmiPodNetwork(vmi *virtv1.VirtualMachineInstance) *virtv1.Network {
 	for _, network := range vmi.Spec.Networks {
 		if network.Pod != nil {
-			return network.Name
+			return &network
 		}
 	}
-	return ""
+	return nil
+}
+
+func vmiPodPrimaryNetworkInterface(vmi *virtv1.VirtualMachineInstance, network virtv1.Network) *virtv1.Interface {
```

Comment:
name it by what it really does 
```suggestion
func vmiNetworkInterface(vmi *virtv1.VirtualMachineInstance, network virtv1.Network) *virtv1.Interface {
```